### ISSUE TITLE
"Specified part does not exist in the package" when provisioning docx in elementFiles

### ISSUE THREAD

**Issue by shurick81** [SharePoint/sp-dev-docs#3830]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

Package is validated in app catalog

#### Observed Behavior

Package is not valid in app catalog. "App Package Error Message" column displays "Invalid SharePoint App package. Error: Specified part does not exist in the package.". Same symptoms when uploading the package into app catalog in SharePoint 2019 or SharePoint Online. In SharePoint 2019 the following messages can be found in URL: https://gist.github.com/shurick81/615042e4f2fba375b893ea22af72e090

#### Steps to Reproduce

1. run in empty directory:
```
yo @microsoft/sharepoint --solution-name "hello-world" --environment "onprem19" --component-type "extension" --extension-type "ApplicationCustomizer" --component-name "HelloWorld" --component-description "HelloWorld extension" --no-insight
```
2. answer `n`
3. In `/hello-world/config/package-solution.json` file add `elementFiles` section, like this:
```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/package-solution.schema.json",
  "solution": {
    "name": "hello-world-client-side-solution",
    "id": "e566aaec-a33f-4df0-889a-d070ba6b7a93",
    "version": "1.0.0.0",
    "includeClientSideAssets": true,
    "features": [
      {
        "title": "Application Extension - Deployment of custom action.",
        "description": "Deploys a custom action with ClientSideComponentId association",
        "id": "ff8a51ad-e8fc-4a9c-a860-aca5cd0e8854",
        "version": "1.0.0.0",
        "assets": {
          "elementManifests": [
            "elements.xml"
          ],
          "elementFiles":[
            "portal.png"
          ]
        }
      }
    ]
  },
  "paths": {
    "zippedPackage": "solution/hello-world.sppkg"
  }
}
```
4. add `portal.png` file to `/hello-world/sharepoint/assets`
5. continue in command line:
```
cd hello-world
npm install
gulp bundle --ship
gulp package-solution --ship
```
6. Upload `/hello-world/sharepoint/solution/hello-world.sppkg` into app catalog and note that it is valid.
7. Delete the app from the app catalog
8. add `portal.docx` file to `/hello-world/sharepoint/assets` directory and a corresponding record to the `/hello-world/config/package-solution.json`:
```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/package-solution.schema.json",
  "solution": {
    "name": "hello-world-client-side-solution",
    "id": "e566aaec-a33f-4df0-889a-d070ba6b7a93",
    "version": "1.0.0.0",
    "includeClientSideAssets": true,
    "features": [
      {
        "title": "Application Extension - Deployment of custom action.",
        "description": "Deploys a custom action with ClientSideComponentId association",
        "id": "ff8a51ad-e8fc-4a9c-a860-aca5cd0e8854",
        "version": "1.0.0.0",
        "assets": {
          "elementManifests": [
            "elements.xml"
          ],
          "elementFiles":[
            "portal.png",
            "portal.docx"
          ]
        }
      }
    ]
  },
  "paths": {
    "zippedPackage": "solution/hello-world.sppkg"
  }
}
```
9. run
```
gulp bundle --ship
gulp package-solution --ship
```
10. Upload the new `/hello-world/sharepoint/solution/hello-world.sppkg` to app catalog.

Now app package is not valid

**Comment by SCL90** [comment]
Hello I reproduced the same problem with css files. is there anyone who has found a solution to this problem ? Thank you !

**Comment by Arceau** [comment]
Hey @SCL90 

I've identified the problem, I haven't come up with a perfect solution to this yet but wanted to give you the information all the same.

The problem is basically that the mime type is not part of the expected contentType within the package-solution gulp task; one option is to manually add it to the file which can be found:

node_modules\@microsoft\sp-build-core-tasks\lib\packageSolution\PackageSolutionTask.js

Just add the following to the bottom of the contentTypes object:

      'css': 'text/css'

This is obviously a bad idea because the packaging will break again on anyone elses machine or if you rebuild your NPM packages.

Another option is to adopt a similar approach to this:
https://github.com/d-martins/spfx-source-maps

It basically packages the solution and then extracts, edits and re-compresses the file with the required mime type.

I've opted for the simple SPFx source code update for now as its only me working on a very simple CSS injection type solution.

If you opt for a more robust solution then it will be great to share as no doubt this will trip up others down the line.

**Comment by MariusHerget** [comment]
Hello,
I can reproduce the bug with a `.docx` file. @Arceau / @shurick81 has one of you developed a solution without manipulating the node modules? Would need something which can be run within a build pipeline on Azure.

**Comment by Arceau** [comment]
Afraid not, I can see 2 options for Azure DevOps, neither of them are particularly pleasing.

The first would be to create a task group that could update the file in the node_modules after the NPM Install, this is obviously pretty brittle and would need to be checked with every new SPFx version.

The other would be to use the Gulp task created by d-martins (linked above), would need some modification to work with your scenario though.

In all honesty, the best move would probably be to not provision files with SPFx, it would probably be easier and more maintainable using a PnP template which can easily be initiated from a pipeline.

**Comment by AJIXuMuK** [comment]
Sorry for the long delay.

We're working on the fix and will include it in SPFx v 1.14

**Comment by AJIXuMuK** [comment]
fixed in 1.14 beta
